File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    public int findCircleNum(int[][] isConnected) {
        boolean[] isVisited = new boolean[isConnected.length];
        int output = 0;
        
        for (int i = 0; i < isVisited.length; i++) {
            if (isVisited[i] == false) {
                output++;
                dfs(isConnected, isVisited, i);
            } 
        }

        return output;
    }

    public void dfs(int[][] isConnected, boolean[] isVisited, int idx) {
        if (isVisited[idx] == true) return;
        else isVisited[idx] = true;
        for (int i = 0; i < isVisited.length; i++) {
            if (isConnected[idx][i] == 1) {
                dfs(isConnected, isVisited, i);
            }
        }
    }
}

File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        visited = [False] * len(isConnected)
        output = 0
        for i in range(len(isConnected)):
            if not visited[i]:
                output += 1
                self.dfs(isConnected, visited, i)
        
        return output

    def dfs(self, isConnected: List[List[int]], visited: List[bool], city: int):
        visited[city] = True
        for i in range(len(isConnected[city])):
            if isConnected[city][i] == 1 and not visited[i]:
                self.dfs(isConnected, visited, i)